### PR TITLE
W-13065850: add automatic module for gateway dependencies

### PR DESCRIPTION
**Pull request by jcgrethe**

### CODE REVIEW

File: modules/all-modules/pom.xml [hunk 1]
```diff
@@ -15,6 +15,7 @@
     </description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.all.modules</javaModuleName>
```
Comment (elrodro83) [inline modules/all-modules/pom.xml:18]: remove

File: modules/memory-management-parent/memory-management-tests/pom.xml [hunk 2]
```diff
@@ -17,6 +17,7 @@
     <description>This module tests the Memory Management Module implementation</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.memory.management.tests</javaModuleName>
```
Comment (elrodro83) [inline modules/memory-management-parent/memory-management-tests/pom.xml:20]: `org.mule.test.memory.management`

File: modules/memory-management-parent/pom.xml [hunk 3]
```diff
@@ -13,6 +13,7 @@
     <description>Modules related to the memory management service</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.memory.management.parent</javaModuleName>
```
Comment (elrodro83) [inline modules/memory-management-parent/pom.xml:16]: remove

File: modules/reboot/pom.xml [hunk 4]
```diff
@@ -12,6 +12,7 @@
     <name>Mule 4 Container</name>
 
     <properties>
+        <javaModuleName>org.mule.runtime.reboot</javaModuleName>
```
Comment (elrodro83) [inline modules/reboot/pom.xml:15]: `org.mule.boot`

File: tests/allure/pom.xml [hunk 5]
```diff
@@ -12,6 +12,7 @@
 	<description>Tests constants to use with Allure Annotations.</description>
 
     <properties>
+		<javaModuleName>org.mule.runtime.test.allure</javaModuleName>
```
Comment (elrodro83) [inline tests/allure/pom.xml:15]: format

File: tests/component-plugin/pom.xml [hunk 6]
```diff
@@ -13,6 +13,7 @@
     <description>Mule test framework components (TCK)</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.component.plugin</javaModuleName>
```
Comment (elrodro83) [inline tests/component-plugin/pom.xml:16]: `org.mule.test.component.plugin `

File: tests/pom.xml [hunk 7]
```diff
@@ -12,6 +12,7 @@
     <name>Mule Tests</name>
 
     <properties>
+        <javaModuleName>org.mule.runtime.mule.test.parent</javaModuleName>
```
Comment (elrodro83) [inline tests/pom.xml:15]: remove

File: tests/test-extensions/classloading-extension/pom.xml [hunk 8]
```diff
@@ -14,6 +14,7 @@
     <name>ClassLoading Extension</name>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.classloading</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/classloading-extension/pom.xml:17]: `org.mule.test.extension.classloading `

File: tests/test-extensions/custom-object-store-extension/pom.xml [hunk 9]
```diff
@@ -15,6 +15,7 @@
     <packaging>jar</packaging>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.custom.object.store</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/custom-object-store-extension/pom.xml:18]: org.mule.test.extension.custom.object.store

File: tests/test-extensions/heisenberg-extension/pom.xml [hunk 10]
```diff
@@ -17,6 +17,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.heisenberg</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/heisenberg-extension/pom.xml:20]: `org.mule.test.extension.heisenberg`

File: tests/test-extensions/implicit-config-extension/pom.xml [hunk 11]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.implicit.config</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/implicit-config-extension/pom.xml:21]: `org.mule.test.extension.implicit.config `

File: tests/test-extensions/implicit-exclusive-config-extension/pom.xml [hunk 12]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.implicit.exclusive.config</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/implicit-exclusive-config-extension/pom.xml:21]: `org.mule.test.extension.implicit.exclusive.config `

File: tests/test-extensions/marvel-extension/pom.xml [hunk 13]
```diff
@@ -17,6 +17,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.marvel</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/marvel-extension/pom.xml:20]: `org.mule.test.extension.marvel`

File: tests/test-extensions/metadata-extension/pom.xml [hunk 14]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.metadata</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/metadata-extension/pom.xml:21]: `org.mule.test.extension.metadata`

File: tests/test-extensions/mule-soap-extension/pom.xml [hunk 15]
```diff
@@ -15,6 +15,7 @@
     <description>A Mule extension for testing Soap Connect</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.soap</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/mule-soap-extension/pom.xml:18]: `org.mule.test.extension.soap `

File: tests/test-extensions/multi-implicit-config-extension/pom.xml [hunk 16]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.implicit.config</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/multi-implicit-config-extension/pom.xml:21]: `org.mule.test.extension.implicit.config`

File: tests/test-extensions/non-implicit-config-extension/pom.xml [hunk 17]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.non.implicit.config</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/non-implicit-config-extension/pom.xml:21]: `org.mule.test.extension.non.implicit.config `

File: tests/test-extensions/parameter-resolver-extension/pom.xml [hunk 18]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.parameter.resolver</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/parameter-resolver-extension/pom.xml:21]: `org.mule.test.extension.parameter.resolver `

File: tests/test-extensions/petstore-extension/pom.xml [hunk 19]
```diff
@@ -17,6 +17,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.petstore</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/petstore-extension/pom.xml:20]: `org.mule.test.extension.petstore`

File: tests/test-extensions/pom.xml [hunk 20]
```diff
@@ -14,6 +14,7 @@
     <name>Parent Pom for Mule Test Extension</name>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extensions.parent</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/pom.xml:17]: remove

File: tests/test-extensions/reconnection-ext/pom.xml [hunk 21]
```diff
@@ -15,6 +15,7 @@
     <description>A Mule extension for testing reconnection</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.reconnection</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/reconnection-ext/pom.xml:18]: `org.mule.test.extension.reconnection`

File: tests/test-extensions/resource-extension/pom.xml [hunk 22]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing that uses the Resource HTTP Service and ResourceLocator</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.resource</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/resource-extension/pom.xml:21]: `org.mule.test.extension.resource `

File: tests/test-extensions/reusable-app/pom.xml [hunk 23]
```diff
@@ -21,6 +21,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.reusable.app</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/reusable-app/pom.xml:24]: `org.mule.test.extension.reusable.app`

File: tests/test-extensions/rick-and-morty-extension/pom.xml [hunk 24]
```diff
@@ -15,6 +15,7 @@
     <description>A Mule extension for testing Soap Connect Custom Transports</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.rick.and.morty</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/rick-and-morty-extension/pom.xml:18]: `org.mule.test.extension.rick.and.morty`

File: tests/test-extensions/sample-data-extension/pom.xml [hunk 25]
```diff
@@ -17,6 +17,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.sample.data</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/sample-data-extension/pom.xml:20]: `org.mule.test.extension.sample.data `

File: tests/test-extensions/semantic-terms-extension/pom.xml [hunk 26]
```diff
@@ -17,6 +17,7 @@
     <description>A Mule extension for testing the semantic terms functionality</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.semantic.terms</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/semantic-terms-extension/pom.xml:20]: `org.mule.test.extension.semantic.terms `

File: tests/test-extensions/some-extension/pom.xml [hunk 27]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.some</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/some-extension/pom.xml:21]: `org.mule.test.extension.some`

File: tests/test-extensions/substitution-group-extension/pom.xml [hunk 28]
```diff
@@ -24,6 +24,7 @@
     </dependencies>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.substitution.group</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/substitution-group-extension/pom.xml:27]: `org.mule.test.extension.substitution.group `

File: tests/test-extensions/subtypes-extension/pom.xml [hunk 29]
```diff
@@ -16,6 +16,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.subtype</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/subtypes-extension/pom.xml:19]: `org.mule.test.extension.subtype`

File: tests/test-extensions/test-config-properties-extension-deprecated-complex/pom.xml [hunk 30]
```diff
@@ -15,6 +15,7 @@
     <description>A Mule extension with complex elements for testing deprecated Configuration Properties provider</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.config.properties.deprecated.complex</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/test-config-properties-extension-deprecated-complex/pom.xml:18]: `org.mule.test.extension.config.properties.deprecated.complex`

File: tests/test-extensions/test-config-properties-extension-deprecated/pom.xml [hunk 31]
```diff
@@ -15,6 +15,7 @@
     <description>A Mule extension for testing deprecated Configuration Properties provider</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.config.properties.deprecated</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/test-config-properties-extension-deprecated/pom.xml:18]: `org.mule.test.extension.config.properties.deprecated`

File: tests/test-extensions/test-config-properties-extension/pom.xml [hunk 32]
```diff
@@ -15,6 +15,7 @@
     <description>A Mule extension for testing Configuration Properties provider</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.config.properties</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/test-config-properties-extension/pom.xml:18]: `org.mule.test.extension.config.properties `

File: tests/test-extensions/test-crafted-extension/pom.xml [hunk 33]
```diff
@@ -15,6 +15,7 @@
     <description>A Mule extension for testing Crafted Connection</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.crafted</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/test-crafted-extension/pom.xml:18]: `org.mule.test.extension.crafted `

File: tests/test-extensions/test-localisation-properties-extension/pom.xml [hunk 34]
```diff
@@ -15,6 +15,7 @@
     <description>A Mule extension for testing Localisation Properties provider</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.localisation</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/test-localisation-properties-extension/pom.xml:18]: `org.mule.test.extension.localisation `

File: tests/test-extensions/test-oauth-extension/pom.xml [hunk 35]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing the OAuth support</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.oauth</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/test-oauth-extension/pom.xml:21]: `org.mule.test.extension.oauth`

File: tests/test-extensions/transactional-extension/pom.xml [hunk 36]
```diff
@@ -17,6 +17,7 @@
     <description>A Mule extension for testing TX support</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.transactional</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/transactional-extension/pom.xml:20]: `org.mule.test.extension.transactional `

File: tests/test-extensions/typed-value-extension/pom.xml [hunk 37]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.typed.value</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/typed-value-extension/pom.xml:21]: `org.mule.test.extension.typed.value `

File: tests/test-extensions/values-extension/pom.xml [hunk 38]
```diff
@@ -17,6 +17,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.values</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/values-extension/pom.xml:20]: `org.mule.test.extension.values `

File: tests/test-extensions/vegan-extension/pom.xml [hunk 39]
```diff
@@ -17,6 +17,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.vegan</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/vegan-extension/pom.xml:20]: `org.mule.test.extension.vegan`

File: tests/test-extensions/weave-function-extension/pom.xml [hunk 40]
```diff
@@ -18,6 +18,7 @@
     <description>A Mule extension for testing</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.extension.weave</javaModuleName>
```
Comment (elrodro83) [inline tests/test-extensions/weave-function-extension/pom.xml:21]: `org.mule.test.extension.weave`

File: tests/test-plugins/dependant-plugin/pom.xml [hunk 41]
```diff
@@ -13,6 +13,7 @@
     <packaging>mule-plugin</packaging>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.plugin.dependant</javaModuleName>
```
Comment (elrodro83) [inline tests/test-plugins/dependant-plugin/pom.xml:16]: `org.mule.test.plugin.dependant`

File: tests/test-plugins/pom.xml [hunk 42]
```diff
@@ -14,6 +14,7 @@
     <description>Provides plugins to use on tests</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.plugins.parent</javaModuleName>
```
Comment (elrodro83) [inline tests/test-plugins/pom.xml:17]: `org.mule.test.plugins.parent`

File: tests/test-policies/pom.xml [hunk 43]
```diff
@@ -14,6 +14,7 @@
     <description>Provides policies to use on tests</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.policies</javaModuleName>
```
Comment (elrodro83) [inline tests/test-policies/pom.xml:17]: `org.mule.test.policies `

File: tests/unit/pom.xml [hunk 44]
```diff
@@ -12,6 +12,7 @@
     <description>Mule Unit Test framework (TCK)</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.test.unit</javaModuleName>
```
Comment (elrodro83) [inline tests/unit/pom.xml:15]: `org.mule.test.unit `

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: modules/tracing/pom.xml [hunk 45]
```diff
@@ -20,6 +20,7 @@
     <description>Modules related to the Runtime Tracer</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.tracing.parent</javaModuleName>
```
Comment (elrodro83) [inline modules/tracing/pom.xml:23]: remove

File: tests/allure/pom.xml [hunk 46]
```diff
@@ -12,7 +12,8 @@
 	<description>Tests constants to use with Allure Annotations.</description>
 
     <properties>
-        <formatterConfigPath>../../formatter.xml</formatterConfigPath>
+		<javaModuleName>org.mule.test.allure</javaModuleName>
+		<formatterConfigPath>../../formatter.xml</formatterConfigPath>
```
Comment (elrodro83) [inline tests/allure/pom.xml:16]: format

File: tests/core-functional-tests/pom.xml [hunk 47]
```diff
@@ -14,6 +14,7 @@
     <description>Module for testing mule core features in a functional way</description>
 
     <properties>
+        <javaModuleName>org.mule.core.functional.tests</javaModuleName>
```
Comment (elrodro83) [inline tests/core-functional-tests/pom.xml:17]: `org.mule.test.core.functional`

**Review by elrodro83 — APPROVED** [review]

**Comment by jcgrethe** [comment]
--validate

**Comment by jcgrethe** [comment]
--validate

**Pull request merged**